firedrake/mesh.py

```diff
@@ -1130,12 +1130,22 @@ def spatial_index(self):
 
         # Calculate the bounding boxes for all cells by running a kernel
         V = functionspace.VectorFunctionSpace(self, "DG", 0, dim=gdim)
-        coords_min = function.Function(V)
-        coords_max = function.Function(V)
+        coords_min = function.Function(V, dtype=utils.RealType)
+        coords_max = function.Function(V, dtype=utils.RealType)
 
         coords_min.dat.data.fill(np.inf)
         coords_max.dat.data.fill(-np.inf)
 
+        if utils.complex_mode:
+            import firedrake.function as function
```

Comment:
We do not need this import? function.Function is imported in line 1123

Reply:
Fixed in c52f12e7975201684676df6250b65ed74e54cc93